File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/GameMapChangeOrientationMessage.scala





















// Generated on 05/03/2013 10:34:39
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameMapChangeOrientationMessage extends MessageDeserializer {
  val messageId = 946

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameMapChangeOrientationMessage.messageId

    val orientation: ActorOrientation.TypeImpl

    def serialize(buf: Buffer) {
      orientation.serialize(buf)
    }

  }

  protected def create(orientation: ActorOrientation.TypeImpl): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val orientation = ActorOrientation.deserialize(buf)

    create(orientation)
  }

}

object GameMapChangeOrientationMessage extends GameMapChangeOrientationMessage {
  case class MessageImpl(orientation: ActorOrientation.TypeImpl) extends Message
  type Target = MessageImpl

  override def create(orientation: ActorOrientation.TypeImpl) = MessageImpl(orientation)

  def apply(orientation: ActorOrientation.TypeImpl) = create(orientation)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameMapChangeOrientationMessage.java


// Generated on 05/08/2013 19:37:43
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameMapChangeOrientationMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 946;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public ActorOrientation orientation;
    
    public GameMapChangeOrientationMessage() { }
    
    public GameMapChangeOrientationMessage(ActorOrientation orientation) {
        this.orientation = orientation;
    }
    
    @Override
    public void serialize(Buffer buf) {
        orientation.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        orientation = new ActorOrientation();
        orientation.deserialize(buf);
    }
    
}
